### PR TITLE
[docker] install git + add missing extensions: zip, pdo_mysql.

### PR DESCRIPTION
**Pull request by aitboudad**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
 FROM php:7.0-apache
 
+RUN apt-get update && apt-get install -y git && apt-get install -y libpng-dev && apt-get install -y zlib1g-dev
+
 # PHP extensions
 RUN docker-php-ext-install mbstring
+RUN docker-php-ext-install mysqli pdo pdo_mysql
```
Comment (teohhanhui) [inline Dockerfile]: This is from my Dockerfile:

```Dockerfile
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        libicu-dev \
        zlib1g-dev \
    && rm -rf /var/lib/apt/lists/* \
    && docker-php-ext-install \
        intl \
        mbstring \
        pdo_mysql \
        zip
```

We definitely don't need `mysqli`, and `pdo` is already included.

Comment (aitboudad) [inline Dockerfile]: fixed

**Comment by dunglas** [comment]
:+1: ping @sroze

File: Dockerfile (hunk 1 above)
Comment (teohhanhui) [inline Dockerfile]: Let's do them in a single `RUN` command. We don't need so many layers...

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,11 @@
 FROM php:7.0-apache
 
+RUN apt-get update && apt-get install -y git && apt-get install -y libpng-dev && apt-get install -y zlib1g-dev
+
 # PHP extensions
 RUN docker-php-ext-install mbstring
+RUN docker-php-ext-install pdo_mysql
```
Comment (teohhanhui) [inline Dockerfile]: Let's do them in a single RUN command. We don't need so many layers...

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,11 @@
 FROM php:7.0-apache
 
+RUN apt-get update && apt-get install -y git && apt-get install -y libpng-dev && apt-get install -y zlib1g-dev
```
Comment (teohhanhui) [inline Dockerfile]: What do we need `libpng-dev` for?

File: Dockerfile (hunk 2 above)
Comment (jderusse) [inline Dockerfile]: Why mysql and not sqlite nor pgsql

File: Dockerfile (hunk 3 above)
Comment (jderusse) [inline Dockerfile]: Exclude `recommands` from install packages

Comment (jderusse) [inline Dockerfile]: Remove /etc/apt/ files after install

**Comment by teohhanhui** [comment]
```Dockerfile
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        libicu-dev \
        zlib1g-dev \
    && rm -rf /var/lib/apt/lists/* \
    && docker-php-ext-install \
        intl \
        mbstring \
        zip
```

File: Dockerfile (hunk 2 above)
Comment (teohhanhui) [inline Dockerfile]: Good point. Let's leave that out.

p/s: I hope to contribute my Docker Compose setup.

Comment (dunglas) [inline Dockerfile]: Currently the default setting of both API Platform and Symfony SE is MySQL. I personally prefer Postgres but MySQL is more used in the PHP world.

**Comment by jderusse** [comment]
Yep, exactly. But remove `*-dev` packages after install because they are only needed for compilation. Replace it by the related lib when needed.

File: Dockerfile (hunk 2 above)
Comment (teohhanhui) [inline Dockerfile]: It's not very useful if mysql is not included in this Docker setup.

**Comment by aitboudad** [comment]
should I include `intl` too ?

**Comment by teohhanhui** [comment]
http://symfony.com/doc/current/reference/requirements.html#optional

**Comment by dzubchik** [comment]
@aitboudad , what about opcache? It gives 2 times less memory consumption and better response time

**Comment by teohhanhui** [comment]
@dzubchik Good idea. And we can install `apcu` too...

```Dockerfile
RUN echo "zend_extension=opcache.so" >> /usr/local/etc/php/conf.d/05-opcache.ini

RUN pecl install \
        apcu \
        apcu_bc-beta
RUN echo "extension=apcu.so" >> /usr/local/etc/php/conf.d/20-apcu.ini \
    && echo "extension=apc.so" >> /usr/local/etc/php/conf.d/21-apc.ini
```

(not sure if `apcu_bc` is necessary...)

@aitboudad @jderusse

**Comment by jderusse** [comment]
Use `docker-php-ext-enable` instead of writing in `.ini` files.

`apcu-bc` is necessary when your application use `apc_*`. I didn't test recently but last versions of Symfony should be compliant.

**Comment by teohhanhui** [comment]
@jderusse If I remember correctly the load order needs to be `apcu.so` followed by `apc.so`, and we have no control over that when using `docker-php-ext-enable`.

**Comment by jderusse** [comment]
You can name `.ini` file with commonly used `zz-` prefix.
https://github.com/docker-library/php/blob/master/7.0/docker-php-ext-enable#L10

**Comment by teohhanhui** [comment]
Didn't know that. Cool!

**Comment by aitboudad** [comment]
I don't have much time to complete this pr, @teohhanhui could you take care of it ? thank's

**Pull request closed without merging**